includes/class-amp-theme-support.php

```diff
@@ -1098,10 +1084,26 @@ public static function prepare_response( $response, $args = array() ) {
 		}
 		$dom = AMP_DOM_Utils::get_dom( $response );
 
-		self::ensure_required_markup( $dom );
+		$xpath = new DOMXPath( $dom );
+
+		// Make sure scripts from the body get moved to the head.
+		$scripts = array();
+		foreach ( $xpath->query( '//body//script[ @custom-element or @custom-template ]' ) as $script ) {
+			$scripts[] = $script;
+		}
+		foreach ( $scripts as $script ) {
+			$head->appendChild( $script );
```

Comment:
The `$head` is undefined. We should grab it with `DOMXPath` and only do the script gathering if it is defined.